### PR TITLE
Feature/lesson4

### PR DESCRIPTION
**Pull request by RozovaOlga**
Lesson 4

### CODE REVIEW

File: src/main/java/lesson4/BubbleSort.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package main.java.lesson4;
+
+/**
+ * BubbleSort
+ */
+public class BubbleSort {
+    public static void main(String[] args) {
+        int a[] = {9, 0, 7, 0, 1, 9, 5, 3, 3,7,0};
```
Comment (DimaGorodkevich1992) [inline src/main/java/lesson4/BubbleSort.java]: magic variable ?

File: src/main/java/lesson4/BubbleSort.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        int temp = 0;
+        /**
+         * Compare the item with the following.If it is larger, the elements change places.
+         * In the last position is the largest element. Exclude the last element from the check and compare again
+         */
+        for (int k = a.length; k > 0; k--) {
```
Comment (DimaGorodkevich1992) [inline src/main/java/lesson4/BubbleSort.java]: magic variable ?

File: src/main/java/lesson4/Number14.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package main.java.lesson4;
+
+/**
+ * Вывести индекс максимальной оценки
+ */
+public class Number14 {
+    public static void main(String[] args) {
+        int massiv[] = {2, 9, 4, 8, 7, 4, 10};
```
Comment (DimaGorodkevich1992) [inline src/main/java/lesson4/Number14.java]: what massiv means ? magic variable again ?

File: src/main/java/lesson4/Number15.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
+package main.java.lesson4;
+
+/**
+ * Определить сумму элементов массива,расположенных между максимальным и минимальным значениями.
+ * Если максимальных и минимальных значений несколько - найменьшее расстояние
+ */
+public class Number15 {
+    public static void main(String[] args) {
+        int massiv[] = {5, 0, 6, 10, 3, 0, 10};
```
Comment (DimaGorodkevich1992) [inline src/main/java/lesson4/Number15.java]: magic variable?

File: src/main/java/lesson4/Number16.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        for (int i = 0; i < massiv.length; i++) {
+            massiv[i] = random.nextInt(10);
+            System.out.print(massiv[i] + " ");
+        }
+        System.out.println();
+        /**
+         * Flip array. Print
+         */
+        int temp;
+        for (int i = 0,j=massiv.length-1; i < massiv.length / 2; i++,j--) {
```
Comment (DimaGorodkevich1992) [inline src/main/java/lesson4/Number16.java]: checkstyle

File: src/main/java/lesson4/Number17.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+/**
+ * Создать двухмерный квадратный массив и заполнить его бабочкой
+ */
+public class Number17 {
+    public static void main(String[] args) {
+        int[][] matrice = new int[10][10];
+        for (int i = 0; i < matrice.length; i++) {
+            for (int j = 0; j < matrice[i].length; j++) {
+                if ((i>j)&&(i<matrice.length/2)) {
```
Comment (DimaGorodkevich1992) [inline src/main/java/lesson4/Number17.java]: checkstyle

File: src/main/java/lesson4/Number17.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+                    matrice[i][j] = 0;
+                }else if ((i>j)&&(j<(matrice[i].length-i-1))) {
```
Comment (DimaGorodkevich1992) [inline src/main/java/lesson4/Number17.java]: checkstyle

File: src/main/java/lesson4/Number17.java [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+                    matrice[i][j] = 0;
+                }else if ((i<j)&&(j>matrice[i].length-i-1)){
+                    matrice[i][j] = 0;
+                }
+                else matrice[i][j]=1;
+                System.out.print(matrice[i][j] + " "); }
+            System.out.println();
+        }
+    }
+}
```
Comment (DimaGorodkevich1992) [inline src/main/java/lesson4/Number17.java]: checkstyle

File: src/main/java/lesson4/Number17.java [hunk 9]
```diff
@@ -0,0 +1,26 @@
+package main.java.lesson4;
+
+/**
+ * Создать двухмерный квадратный массив и заполнить его бабочкой
+ */
+public class Number17 {
+    public static void main(String[] args) {
+        int[][] matrice = new int[10][10];
+        for (int i = 0; i < matrice.length; i++) {
```
Comment (DimaGorodkevich1992) [inline src/main/java/lesson4/Number17.java]: matrix != matrice

File: src/main/java/lesson4/String/Stroka.java [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+ * Перевернуть строку
+ */
+
+public class Stroka {
+    public static void main(String[] args) {
+        String string="Hello";
+        /**
+         * Create an array of string characters
+         */
+        char[] massivChar = string.toCharArray();
```
Comment (DimaGorodkevich1992) [inline src/main/java/lesson4/String/Stroka.java]: var name

File: src/main/java/lesson4/String/Stroka.java [hunk 11]
```diff
@@ -0,0 +1,27 @@
+package main.java.lesson4.String;
+
+/**
+ * Перевернуть строку
+ */
+
+public class Stroka {
+    public static void main(String[] args) {
+        String string="Hello";
```
Comment (DimaGorodkevich1992) [inline src/main/java/lesson4/String/Stroka.java]: checkstyle

File: src/main/java/lesson4/String/Stroka.java [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+        /**
+         * Create an array of string characters
+         */
+        char[] massivChar = string.toCharArray();
+        /**
+         * Flip Array. Print
+         */
+        char temp;
+        for (int i = 0,j=string.length()-1; i < massivChar.length / 2; i++,j--) {
```
Comment (DimaGorodkevich1992) [inline src/main/java/lesson4/String/Stroka.java]: checkstyle and very complicated for

Comment (RozovaOlga) [reply src/main/java/lesson4/String/Stroka.java]: fixed everything

File: src/main/java/lesson4/BubbleSort.java (hunk 2 above)
Comment (RozovaOlga) [reply src/main/java/lesson4/BubbleSort.java]: don't understand the mistake

Comment (DimaGorodkevich1992) [reply src/main/java/lesson4/BubbleSort.java]: what is "a" ?

**Pull request merged**